File: app/src/main/java/se/appshack/android/refactoring/PokemonListResponse.java
package se.appshack.android.refactoring;

import com.google.gson.annotations.SerializedName;

import java.util.List;

class PokemonListResponse {

    @SerializedName("count")
    public int count;

    @SerializedName("next")
    public String next;

    @SerializedName("previous")
    public String previous;

    @SerializedName("results")
    public List<NamedResponseModel> results;
}

File: app/src/main/java/se/appshack/android/refactoring/main/data/model/response/PokemonListResponse.kt
package se.appshack.android.refactoring.main.data.model.response

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import se.appshack.android.refactoring.main.data.model.NamedResponseModel

@Parcelize
data class PokemonListResponse (

    @SerializedName("count")
    var count: Int = 0,

    @SerializedName("next")
    var next: String? = null,

    @SerializedName("previous")
    var previous: String? = null,

    @SerializedName("results")
    var results: List<NamedResponseModel>? = null

): Parcelable

